apisix/discovery/kubernetes/init.lua

```diff
@@ -434,6 +442,9 @@ local function multiple_mode_init(confs)
         end
 
         local endpoint_dict = ngx.shared["kubernetes-" .. id]
+        if not is_http then
+            endpoint_dict = ngx.shared["kubernetes-" .. id .. "-stream"]
+        end
```

Comment:
It would be better to refactor these three changes into a function, instead of three if-else blocks.